modules/misc/xdg-mime.nix

```diff
@@ -0,0 +1,45 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.xdg.mime;
+
+in
+
+{
+  options = {
+    xdg.mime.enable = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether to install files to support the
+        <link xlink:href="https://specifications.freedesktop.org/shared-mime-info-spec/shared-mime-info-spec-latest.html">XDG Shared MIME-info specification</link> and the
+        <link xlink:href="https://specifications.freedesktop.org/mime-apps-spec/mime-apps-spec-latest.html">XDG MIME Applications specification</link>.
+      '';
+    };
+  };
+
+  config = mkIf config.xdg.mime.enable {
+    home.packages = [
+      # this package also installs some useful data, as well as its utilities
+      pkgs.shared-mime-info
+    ];
+
+    home.sessionVariables = {
+      XDG_DATA_DIRS = "$XDG_DATA_DIRS:${config.home.profileDirectory}/share";
```

Comment:
I believe this belongs in `nix.sh` or some place where it is set for all people using Nix on non-NixOS, not just people using Home Manager.

Reply:
I tend to agree; unfortunately Eelco seems to think otherwise: https://github.com/NixOS/nix/pull/2443
Perhaps the simplest solution here would be to wrap just `xdg-open` etc. instead in nixpkgs, analogously to https://github.com/NixOS/nixpkgs/pull/54525?

Reply:
Fair enough, but it should only be set for people on non-NixOS since otherwise the `${config.home.profileDirectory}/share` will show up twice. I would suggest to remove this for now and I'll make sure to re-add it in a later commit once I've figured out how to do it nicely. I would also place this somewhere else, perhaps in `misc/xdg.nix` since the variable isn't specifically related to xdg-mime.

Reply:
Ok, removed. I'll open a PR against `xdg-open` in nixpkgs to include `XDG_DATA_DIRS` there for everyone.

Reply:
I'm not sure how to do this in a nice way in a xdg-open wrapper since you can't at package build time know where the user profile will be found.

Reply:
Ah, you're right of course.